code/modules/paperwork/fax.dm

```diff
@@ -110,56 +231,155 @@
 	switch(action)
 		// Pulls the paper out of the fax machine
 		if("remove")
-			var/obj/item/paper/paper = paper_contain
-			paper.forceMove(drop_location())
-			paper_contain = null
+			var/obj/item/loaded = loaded_item?.resolve()
+			if (!loaded)
+				return
+			loaded.forceMove(drop_location())
+			loaded_item = null
 			playsound(src, 'sound/machines/eject.ogg', 50, FALSE)
 			update_appearance()
 			return TRUE
 		if("send")
-			if(send(paper_contain, params["id"]))
-				log_paper("[usr] has sent a fax with the message \"[paper_contain.get_raw_text()]\" to [params["id"]].")
-				paper_contain = null
+			var/obj/item/loaded = loaded_item?.resolve()
+			if (!loaded)
+				return
+			var/destination = params["id"]
+			if(send(loaded, destination))
+				log_fax(loaded, destination, params["name"])
+				loaded_item = null
 				update_appearance()
 				return TRUE
 		if("history_clear")
 			history_clear()
 			return TRUE
 
+/**
+ * Records logs of bureacratic action
+ * Arguments:
+ * * sent - The object being sent
+ * * destination_id - The unique ID of the fax machine
+ * * name - The friendly name of the fax machine, but these can be spoofed so the ID is also required
+ */
+/obj/machinery/fax/proc/log_fax(obj/item/sent, destination_id, name)
+	if (istype(sent, /obj/item/paper))
+		var/obj/item/paper/sent_paper = sent
+		log_paper("[usr] has sent a fax with the message \"[sent_paper.get_raw_text()]\" to [name]/[destination_id].")
+		return
+	if (istype(sent, /obj/item/throwing_star))
+		log_attack("[usr] has faxed [sent] to [name]/[destination_id].]")
+		return
+	log_game("[usr] has faxed [sent] to [name]/[destination_id].]")
```

Comment:
logging in 2 different places for faxing seems too confusing for admins to look through